app/Services/Course/UpdateService.php

```diff
@@ -20,41 +27,96 @@ public function __invoke(
+                ]
+            );
 
-        $course->courseDeadline()->updateOrCreate(
-            ['course_id' => $course->id],
-            [
-                'fixed_date' => $deadlineType === DeadlineTypeEnum::FIXED_DATE ? $fixedDate : null,
-                'relative_days' => $deadlineType === DeadlineTypeEnum::RELATIVE_DAYS ? $relativeDays : null,
-            ]
-        );
+            // 受講期限タイプに応じて処理を分岐
+            match ($deadlineType) {
+                DeadlineTypeEnum::FIXED_DATE => $this->updateFixedDeadline($course, $fixedDate),
+                DeadlineTypeEnum::RELATIVE_DAYS => $this->updateRelativeDeadline($course, $relativeDays, $calculateDeadline),
+                // deadline が NONE の場合は受講生の期限を null に更新
+                DeadlineTypeEnum::NONE => $course->attendances()->update(['attendance_deadline' => null]),
+                // 想定外の受講期限タイプの場合は例外をスロー
+                default => throw new LogicException("Unexpected deadline type: {$deadlineType->value}"),
+            };
     }
-
     /**
-     * 受講期限設定があるかどうか
+     * 受講期限（固定年月日）の更新
      */
-    private function hasDeadline(DeadlineTypeEnum $deadlineType): bool
+    private function updateFixedDeadline(Course $course, ?string $fixedDate): void
     {
-        return in_array($deadlineType, [
-            DeadlineTypeEnum::FIXED_DATE,
-            DeadlineTypeEnum::RELATIVE_DAYS,
-        ], true);
+        // 固定年月日が null の場合は「nullではダメ」とスロー
+        if ($fixedDate === null) {
+            throw new InvalidArgumentException('Fixed date cannot be null.');
+        }
+        // 受講生の受講期限（固定日）を一括更新
+        $attendanceIds = $course->attendances()->update([
+            'attendance_deadline' => $fixedDate
+        ]);
+    }
+
+    /**
+     * 受講期限（受講日から〇日）の更新
+     */
+    private function updateRelativeDeadline(
+        Course $course,
+        ?int $relativeDays,
+        CalculateDeadlineService $calculateDeadline,
+    ): void{
+        // 受講日からの日数がnullの場合は「nullではダメ」とスロー
+        if ($relativeDays === null) {
+            throw new InvalidArgumentException('Relative days must not be null.');
+        }
+        
+        // 受講期限を計算して更新
+        $attendances = $course->attendances()->get();
+        foreach ($attendances as $attendance) {
+            // 受講開始日を取得
+            $startAt = new DateTimeImmutable($attendance->created_at);
+            // 新しい受講期限を計算
+            $newDeadline = $calculateDeadline(
+                DeadlineTypeEnum::RELATIVE_DAYS->value,
+                null,
+                $relativeDays,
+                $startAt
+            );
+
+            // 新しい受講期限が null の場合は例外をスロー
+            if ($newDeadline === null) {
+                throw new RuntimeException('Failed to calculate new attendance deadline.');
+            }
+
+            // 受講生の受講期限を更新
+            $attendance->update([
+                'attendance_deadline' => $newDeadline?->format('Y-m-d'),
```

Comment:
メッセージ通り、前述のコードでnullである場合の処理は行なっているので、nullsafeは削除するようにお願いします！

```suggestion
                'attendance_deadline' => $newDeadline->format('Y-m-d'),
```

Reply:
作業会でも教えていただいたとおり、修正しました。
ありがとうございます。